tasks/Task.cpp

```diff
@@ -74,6 +74,23 @@ bool Task::configureHook()
         return false;
     }
 
+    // creating dynamic ports
+    outputports = _outputPorts.get();
+    for (size_t i = 0; i < outputports.size(); ++i)
+    {
+        canbus::CanOutputPort const& outputport(outputports[i]);
+        if (ports()->getPort(outputport.ports_name))
+        {
+            RTT::log(RTT::Error) << "output port " <<  outputport.ports_name << " is listed more than once in the outputs configuration property" << RTT::endlog();
+            return false;
+        }
+        RTT::OutputPort<canbus::Message>* new_output_port = new RTT::OutputPort<canbus::Message>(outputport.ports_name);
+        ports()->addPort(outputport.ports_name, *new_output_port);
+        // register the mapping
+        Mapping mapping = { outputport.ports_name, outputport.id & outputport.mask, outputport.mask, new_output_port };
```

Comment:
It would be a lot simpler for the cleanup to add a 'remove_on_cleanup' boolean flag to Mapping and set it to true here (and false in the operation). Cleanup would only have to remove the ports that have the flag set to true.